pkg/controller/direct/sql/merge.go

```diff
@@ -626,7 +626,7 @@ func MergeDesiredSQLInstanceWithActual(desired *krm.SQLInstance, refs *SQLInstan
 	}
 
 	if desired.Spec.Settings.DiskAutoresize != nil {
-		if desired.Spec.Settings.DiskAutoresize != actual.Settings.StorageAutoResize {
+		if direct.ValueOf(desired.Spec.Settings.DiskAutoresize) != direct.ValueOf(actual.Settings.StorageAutoResize) {
 			// Change disk autoresize
 			updateRequired = true
```

Comment:
We might also be able to look at the updateTime...

Reply:
It felt like this was a more reliable way to do it, but am planning to switch to use the mapper / generator code soon anyway.

Reply:
Though maybe the mapper generator code wouldn't really help here. Can you elaborate a bit on what you mean by using updateTime? Not sure I understand completely.